src/client/src/rt-components/platform/adapters/openfin/openFin.ts

```diff
@@ -4,25 +4,35 @@ import { openDesktopWindow } from './window'
 import { fromEventPattern } from 'rxjs'
 import { excelAdapter } from './excel'
 import { CurrencyPairPositionWithPrice } from 'rt-types'
-
-export const openFinNotifications: any[] = []
-
-declare const window: Window & { onNotificationMessage: any }
-
-export const setupGlobalOpenfinNotifications = () => {
-  if (typeof fin !== 'undefined' && !window.onNotificationMessage) {
-    // openfin requires a global onNotificationMessage function to be defined before its notification structure is initialized in the platform adapter.
-    // NotificationRoute is imported lazily, thus we cannot define the function in that file. (Testing has shown it's already too late.)
-    // - D.S.
-    window.onNotificationMessage = (message: any) => openFinNotifications.push(message)
-  }
-}
+import {
+  Notification,
+  NotificationButtonClickedEvent,
+  NotificationOptions,
+} from 'openfin-notifications'
+import { NotificationMessage } from '../browser/utils/sendNotification'
 
 type OpenFinWindowState = Parameters<Parameters<fin.OpenFinWindow['getState']>[0]>[0]
 export default class OpenFin extends BasePlatformAdapter {
   readonly name = 'openfin'
   readonly type = 'desktop'
 
+  private createNotification: (options: NotificationOptions) => Promise<Notification>
+
+  constructor() {
+    super()
+    this.createNotification = require('openfin-notifications').create
+
+    require('openfin-notifications').addEventListener(
```

Comment:
Is there a reason for the inline `require` rather than importing at the top of the file?

Reply:
Yes, we need the inline require, otherwise the browser version is not working. Fyi, I have this error
`ReferenceError: fin is not defined
(anonymous function)
C:/git/ReactiveTraderCloud/src/client/node_modules/openfin-notifications/dist/client/main-bundle.js:535
  532 | 
  533 | exports.eventEmitter = new events_1.EventEmitter();
  534 | 
> 535 | if (fin.Window.me.uuid !== internal_1.SERVICE_IDENTITY.uuid || fin.Window.me.name !== internal_1.SERVICE_IDENTITY.name) {
      | ^  536 |   exports.channelPromise = typeof fin === 'undefined' ? Promise.reject(new Error('fin is not defined. The openfin-notifications module is only intended for use in an OpenFin application.')) : fin.InterApplicationBus.Channel.connect(internal_1.SERVICE_CHANNEL, {
  537 |     payload: {
  538 |       version: '0.10.1'
View compiled
__webpack_require__
C:/git/ReactiveTraderCloud/src/client/node_modules/openfin-notifications/dist/client/main-bundle.js:62`

However, I improved a bit the code to at least not repeat the require